From 0634eb9e2bf47b028988e1932fa444f56f49332a Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Wed, 23 Sep 2015 15:32:35 +0000
Subject: [PATCH 4/4] Set the appropriate locale

By calling `setlocale(LC_ALL, "");`, we give the user a chance to
override the ASCII-only default so that non-ASCII arguments are
converted into UTF-16 properly, which is required to pass them
correctly to the actual Win32 executable.

This is necessary because POSIX dictates that we start up with the C
locale, ignoring all environment variables, until that `setlocale()`
call.

This fixes https://github.com/git-for-windows/git/issues/412

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 unix-adapter/main.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/unix-adapter/main.cc b/unix-adapter/main.cc
index 80d01a2..14fabbf 100644
--- a/unix-adapter/main.cc
+++ b/unix-adapter/main.cc
@@ -344,6 +344,7 @@ int main(int argc, char *argv[])
         return 0;
     }
 
+    setlocale(LC_ALL, "");
     setupWin32Environment();
 
     winsize sz;
-- 
2.5.2

